### ISSUE TITLE
value asInstanceOf is not a member of <notype> when using refined NonEmptyString 

### ISSUE THREAD

**Issue by rolman243** [andyglow/scala-jsonschema#58]
I'm seeing this error when trying to generate json schema for case class using refined NonEmptyString type:
```
value asInstanceOf is not a member of <notype>
Note that <none> extends Any, not AnyRef.
Such types can participate in value classes, but instances
cannot appear in singleton types or in reference comparisons.
```
When changing NonEmptyString to String everything works fine

example code:
```
import com.github.andyglow.json.JsonFormatter
import com.github.andyglow.jsonschema.AsValue
import com.github.andyglow.jsonschema.RefinedSupport._
import eu.timepit.refined.types.string.NonEmptyString
import json.{Json, Schema}

case class MyTest(string: NonEmptyString)

implicit val myTestSchema: Schema[MyTest] = Json.schema[MyTest]

println(JsonFormatter.format(AsValue.schema(myTestSchema, json.schema.Version.Draft04())))
```

I'm using scala 2.13.3 with scala-jsonschema 0.2.8

I thought NonEmptyString should work even not being mentioned in the ReadMe explicitly but
eu.timepit.refined.boolean.Not and eu.timepit.refined.collection.Empty are supported.

**Comment by andyglow** [comment]
hey, @rolman243 
thanks for reporting it. let me check if it wasn't trivially forgotten

**Comment by andyglow** [comment]
yeah, support for that refinements wasn't added initially
this has to fix the issue
https://github.com/andyglow/scala-jsonschema/commit/e01aee668c90a5d86eeb42157344eb7bb8586993
please check out next 0.2.9 version (has to become available in next few hours)

**Comment by rolman243** [comment]
Hey @andyglow,
thank you for having a look at it so fast.
I'll check it out tomorrow.

**Comment by rolman243** [comment]
It works now @andyglow but the 0.2.9 release is not available on maven, I cloned the project and built locally.
Many thanks for fast fix

**Comment by rolman243** [comment]
Hey @andyglow,
do you have any time for releasing 0.2.9, it's still not available on maven?

**Comment by andyglow** [comment]
oh.. so sorry. i somehow released only for 2.11 back then.
fixed that. must be available in few hours.

**Comment by rolman243** [comment]
Thank you @andyglow for the great work
